include/boost/basic_any.hpp

```diff
@@ -0,0 +1,440 @@
+        template<typename ValueType>
+        basic_any(ValueType&& value
+            , typename boost::disable_if<boost::is_same<basic_any&, ValueType> >::type* = 0 // disable if value has type `basic_any&`
+            , typename boost::disable_if<boost::is_const<ValueType> >::type* = 0) // disable if value has type `const ValueType&&`
+          : man(0), content()
+        {
+            content.large_value = 0;
+            create(*this, static_cast<ValueType&&>(value), is_small_object<BOOST_DEDUCED_TYPENAME boost::decay<ValueType>::type>());
+        }
+#endif
+
+        ~basic_any() BOOST_NOEXCEPT
+        {
+            if (man)
+            {
+                man(Destroy, *this, 0, 0);
+            }
+        }
+
+    public: // modifiers
+
+        basic_any & swap(basic_any & rhs) BOOST_NOEXCEPT
+        {
+            if (this != &rhs)
+            {
+                if (man && rhs.man)
+                {
+                    basic_any tmp;
+                    rhs.man(Move, tmp, &rhs, 0);
+                    man(Move, rhs, this, 0);
+                    tmp.man(Move, *this, &tmp, 0);
+                }
+                else if (man)
+                {
+                    man(Move, rhs, this, 0);
+                }
+                else if (rhs.man)
+                {
+                    rhs.man(Move, *this, &rhs, 0);
+                }
+            }
+            return *this;
+        }
+
+
+#ifdef BOOST_NO_CXX11_RVALUE_REFERENCES
+        template<typename ValueType>
+        basic_any & operator=(const ValueType & rhs)
+        {
+            basic_any(rhs).swap(*this);
+            return *this;
+        }
+
+        basic_any & operator=(basic_any rhs)
+        {
+            rhs.swap(*this);
+            return *this;
+        }
+
+#else 
+        basic_any & operator=(const basic_any& rhs)
+        {
+            basic_any(rhs).swap(*this);
+            return *this;
+        }
+
+        // move assignment
+        basic_any & operator=(basic_any&& rhs) BOOST_NOEXCEPT
+        {
+            rhs.swap(*this);
+            basic_any().swap(rhs);
+            return *this;
+        }
+
+        // Perfect forwarding of ValueType
+        template <class ValueType>
+        basic_any & operator=(ValueType&& rhs)
+        {
+            basic_any(static_cast<ValueType&&>(rhs)).swap(*this);
+            return *this;
+        }
+#endif
+
+    public: // queries
+
+        bool empty() const BOOST_NOEXCEPT
+        {
+            return !man;
+        }
+
+        void clear() BOOST_NOEXCEPT
+        {
+            basic_any().swap(*this);
+        }
+
+        const boost::typeindex::type_info& type() const BOOST_NOEXCEPT
+        {
+            return man
+                    ? *static_cast<const boost::typeindex::type_info*>(man(Typeinfo, const_cast<basic_any&>(*this), 0, 0))
+                    : boost::typeindex::type_id<void>().type_info();
+        }
+
+#ifndef BOOST_NO_MEMBER_TEMPLATE_FRIENDS
```

Comment:
Kill that. Let's assume that BOOST_NO_MEMBER_TEMPLATE_FRIENDS is never defined.

Reply:
done